src/main/java/ch/njol/skript/lang/SkriptParser.java

```diff
@@ -263,7 +263,30 @@ private <T extends SyntaxElement> T parse(Iterator<? extends SyntaxElementInfo<?
 								x = x2;
 							}
 							T t = info.c.newInstance();
-							if (t.init(res.exprs, i, getParser().getHasDelayBefore(), res)) {
+							ParserInstance parser = getParser();
+							List<Class<? extends Structure>> usableStructures = t.getUsableStructures();
+							if (!usableStructures.isEmpty()) {
+								if (parser.getCurrentStructure() == null) {
+									Skript.error("'" + res.expr + "' cannot be used outside of the allowed structures.");
+									continue;
```

Comment:
break the while loop here. Using continue will print the error multiple times if multiple syntaxes pass the pattern. Either print error log and return null or break out of the while loop. Same goes with the continues below.

Reply:
Continue should be fine here because it follows the same behavior as `init`, which is to continue onto the next pattern